core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java

```diff
@@ -207,93 +207,91 @@ private Optional<ExecutedVertex> prepareInternal(
     final LedgerHeader parentHeader = vertex.getParentHeader().getLedgerHeader();
     final AccumulatorState parentAccumulatorState = parentHeader.getAccumulatorState();
 
-    // The `commit()` operation may concurrently advance the ledger, so let's only read out the
-    // (immutable) accumulator state once.
-    // (we do not rely on any other state, and the Rust side of the prepare operation is properly
-    // synchronized there, which means we can carry on without the lock).
-    final AccumulatorState againstAccumulatorState;
+    final StateComputerResult result;
     synchronized (this.commitAndAdvanceLedgerLock) {
-      againstAccumulatorState = this.currentLedgerHeader.getHeader().getAccumulatorState();
-    }
+      final AccumulatorState againstAccumulatorState =
```

Comment:
_(a note from the author)_

In theory, the java prepare could stay with that short lock, and we could handle the "oops, old state" case in rust (i.e. return such result gracefully). But @malexandru-rdx convinced me that it is not worth it.